### PR TITLE
parts: add craftctl support

### PR DESCRIPTION
**Pull request by cmatsuoka**
Add craftctl wrapper to execute the craft-parts utility from the
snapcraft snap, and define project variables that can be set
using craftctl. Add `adopt-info` to project configuration.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----
CRAFT-964

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -135,3 +135,9 @@ parts:
         snapcraftctl set-grade "$grade"
         ln -sf ../usr/bin/python3.8 $SNAPCRAFT_PART_INSTALL/bin/python3
     after: [snapcraft-libs]
+
+  craftctl:
```
Comment (sergiusens) [inline snap/snapcraft.yaml:139]: this should probably be a script entry in setup.py

File: libexec/craftctl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec "$SNAP"/bin/python "$SNAP"/bin/craftctl "$@"
```
Comment (sergiusens) [inline libexec/craftctl:3]: I would put /snap/snapcraft/current/bin/craftctl in libexec/snapcraft/craftctl and have snapcraft update the shebang instead

Comment (cmatsuoka) [reply libexec/craftctl]: Updated to move the actual python script to `libexec/snapcraft/craftctl` and replace the shebang to python in snap current.

**Review by sergiusens — APPROVED** [review]

**Comment by cmatsuoka** [comment]
TODO: change core22 spread test layout and use ubuntu-22.04 where appropriate

**Pull request merged**